software/deploy/deploy_test.go

```diff
@@ -1017,3 +1017,81 @@ func (s *Suite) TestDeployDagsOnlyFailure() {
 		s.True(os.IsNotExist(err))
 	})
 }
+
+func (s *Suite) TestUpdateDeploymentImage() {
+	testUtil.InitTestConfig(testUtil.SoftwarePlatform)
+	deploymentID := "test-deployment-id"
+	wsID := "test-workspace-id"
+	runtimeVersion := "12.1.1"
+	imageName := "imageName"
+	releaseName := "releaseName"
+
+	s.Run("When runtimeVersion is empty", func() {
+		houstonMock := new(houston_mocks.ClientInterface)
+		returnedDeploymentID, err := UpdateDeploymentImage(houstonMock, deploymentID, wsID, "", imageName)
+		s.ErrorIs(err, ErrRuntimeVersionNotPassedForRemoteImage)
+		houstonMock.AssertExpectations(s.T())
```

Comment:
Houston mock assertion should be part of the suite's `TearDownSubTest` if it is being called at the end of every test